File: Grocerly.Hybrid/Grocerly.Hybrid/Grocerly.Hybrid/Services/Interfaces/IShoppingListService.cs
using Grocerly.Hybrid.Models;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace Grocerly.Hybrid.Services.Interfaces
{
    public interface IShoppingListService
    {
        Task<IEnumerable<ShoppingList>> GetShoppingListsWithStatus(Status status);
        Task<IEnumerable<ShoppingListItem>> GetProductsForShoppingList(Guid id);
        Task<ShoppingList> CreateShoppingList();
        Task<Product> AddProductToList(Guid product_id, Guid shoppingList_id);
        Task<IEnumerable<ShoppingList>> GetShoppingListsForUser(Guid id, Status status);
        Task<bool> UpdateShoppingList(ShoppingList shoppingList);
        Task<ShoppingList> GetShoppingList(Guid id);
    }
}

File: Grocerly.Android/app/src/main/java/grocerly/fhict/com/grocerly/services/interfaces/IShoppingListService.java
package grocerly.fhict.com.grocerly.services.interfaces;

import java.util.List;

import grocerly.fhict.com.grocerly.models.Product;
import grocerly.fhict.com.grocerly.models.ShoppingList;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface IShoppingListService {
    @GET("api/shoppinglist/{id}/products")
    Call<List<Product>> getShoppingListItems(@Path("id") String id);

    @GET("api/shoppinglist/")
    Call<List<ShoppingList>> allShoppingLists();

}
